### PR TITLE
Initial set up

### PR DESCRIPTION
**Pull request by simpikkle**

### CODE REVIEW

File: ui/src/components/categories/CategoriesList.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import MenuItem from "@material-ui/core/MenuItem";
+import MenuList from "@material-ui/core/MenuList";
+
+export const CategoriesList = (props) => {
+
+    let getMenuItems = () => props.categories.map((category, i) => {
```
Comment (alxb1t) [inline ui/src/components/categories/CategoriesList.js]: No need for a function here.
You can create the collection and use in JSX.
`
const menuItems = props.categories.map((category, i) => {//TODO});
`

File: ui/src/components/forms/AddQuestionForm.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            difficulty: this.state.difficulty,
+        });
+        this.setState({
+            question: "",
+            answer: "",
+            difficulty: 0
+        })
+    };
+
+    saveQuestion = question => {
```
Comment (alxb1t) [inline ui/src/components/forms/AddQuestionForm.js]: We can move it to a separate class which can send requests and here use this class.

File: ui/src/components/forms/AddQuestionForm.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    handleChange = fieldName => {
+        return event => {
+            this.setState({
+                [fieldName]: event.target.value
+            });
+        };
+    };
+
+    save = () => {
```
Comment (alxb1t) [inline ui/src/components/forms/AddQuestionForm.js:43]: Add a progress bar in order to show to user that the request is being processed https://material-ui.com/components/progress/

File: ui/src/components/forms/AddQuestionForm.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        this.saveQuestion({
+            question: this.state.question,
+            answer: this.state.answer,
+            category: this.state.category,
+            difficulty: this.state.difficulty,
+        });
+        this.setState({
```
Comment (alxb1t) [inline ui/src/components/forms/AddQuestionForm.js]: Reset the form after the response was successful.

File: ui/src/components/forms/AddQuestionForm.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            },
+            body: JSON.stringify(question),
+        })
+            .then(response => {
+                return response.json();
+            })
+            .catch(error => console.error(error));
+    };
+
+    categories = () => {
```
Comment (alxb1t) [inline ui/src/components/forms/AddQuestionForm.js]: To be more cool, we can use
`categories = () => this.props.categories.map((category, i) => <MenuItem key={i} value={i}>{category}</MenuItem>);`

File: ui/src/components/forms/QuestionsForm.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    selectCategory = (category) => {
+        this.setState({
+            currentCategory: category
+        })
+    };
+
+    componentDidMount() {
+        fetch(`${BASE_URL}/categories`)
```
Comment (alxb1t) [inline ui/src/components/forms/QuestionsForm.js:30]: Move to a request api class.

File: ui/src/components/forms/QuestionsForm.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    selectTab(event, tab) {
+        debugger;
+        this.setState({
+            tab: tab
+        })
+    }
+
+    render() {
+
+        let elements = {
```
Comment (alxb1t) [inline ui/src/components/forms/QuestionsForm.js:49]: Move outside of render.

File: ui/src/components/questions/QuestionItem.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import {Typography} from "@material-ui/core";
+import Grid from "@material-ui/core/Grid";
+import Avatar from "@material-ui/core/Avatar";
+import "../../css/App.css";
+import Button from "@material-ui/core/Button";
+
+export const QuestionItem = (props) => {
+
+    let answerText = props.questionItem.answer.split('\n').map((item, i) => {
```
Comment (alxb1t) [inline ui/src/components/questions/QuestionItem.js:10]: Rende Raw html.

File: ui/src/components/questions/QuestionsList.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        this.fetchQuestions(nextProps.currentCategory);
+        if (this.props.currentCategory !== nextProps.currentCategory) {
+            this.setState({
+                selectedQuestion: null
+            })
+        }
+    }
+
+    fetchQuestions(categoryId) {
+        fetch(`${BASE_URL}/questions/${categoryId}`)
```
Comment (alxb1t) [inline ui/src/components/questions/QuestionsList.js]: Move to api service.

File: ui/src/components/questions/QuestionItem.js (hunk 8 above)
Comment (simpikkle) [reply ui/src/components/questions/QuestionItem.js:10]: Couldn't solve it fast, will make it in seaprate PR

**Pull request merged**